pinot-spi/src/test/resources/segment-metadata-with-version-header.properties

```diff
@@ -0,0 +1,125 @@
+# segment.metadata.version=version1
```

Comment:
Why is it start from `version1`?
I'd suggest treating the one without header as version 1, and the new format starts as version 2.
Also suggest simplifying it as `# version=2`. This is a general setting for all configurations, instead of just the segment metadata